Framework/Interaction/MetaData/MetaDataObject.php

```diff
@@ -169,7 +169,7 @@ public function validateData(array $data)
 
         foreach ($this->requiredRules as $requiredRule) {
             if (!array_key_exists($requiredRule->getName(), $validatedData) ||
-                empty($validatedData[$requiredRule->getName()]) ||
+                (empty($validatedData[$requiredRule->getName()]) && $validatedData[$requiredRule->getName()] != 0) ||
                 is_null($validatedData[$requiredRule->getName()])) {
```

Comment:
please, consider omitting using empty,becayse array_key_exists already check $validateData on key existing

Reply:
replaced by using the \Zend\Validator\NotEmpty